tests/baselines/reference/parenthesizedContexualTyping1.types

```diff
@@ -220,11 +220,11 @@ var l = fun(((Math.random() < 0.5 ? ((x => x)) : ((x => undefined)))), ((x => x)
 >x => undefined : (x: number) => any
 >x : number
 >undefined : undefined
->((x => x)) : (x: number) => number
->(x => x) : (x: number) => number
->x => x : (x: number) => number
->x : number
->x : number
+>((x => x)) : (x: any) => any
+>(x => x) : (x: any) => any
+>x => x : (x: any) => any
+>x : any
+>x : any

```

Comment:
Oh no - is this #241? This is definitely a regression.

To clarify in readability for anyone jumping in:

```TypeScript
function fun<T>(g: (x: T) => T, h: (y: T) => T, x: T): T;
```

Our parameters are

* `((Math.random() < 0.5 ? ((x => x)) : ((x => undefined))))`
* `((x => x))`
* `10`

Clearly this is not desirable.